aws/resource_aws_cloudsearch_service.go

```diff
@@ -0,0 +1,471 @@
+		Update: resourceAwsCloudSearchDomainUpdate,
+		Delete: resourceAwsCloudSearchDomainDelete,
+
+		Schema: map[string]*schema.Schema{
+			"domain_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"instance_type": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"replication_count": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				Default:  0,
+			},
+			"partition_count": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				Default:  1,
+			},
+			"indexes": {
+				Type:     schema.TypeList,
+				Required: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"type": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"search": {
+							Type:     schema.TypeBool,
+							Required: true,
+						},
+						"facet": {
+							Type:     schema.TypeBool,
+							Optional: true,
+						},
+						"return": {
+							Type:     schema.TypeBool,
+							Required: true,
+						},
+						"sort": {
+							Type:     schema.TypeBool,
+							Optional: true,
+						},
+						"highlight": {
+							Type:     schema.TypeBool,
+							Optional: true,
+						},
+						"analysis_scheme": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"default_value": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+					},
+				},
+			},
+			"access_policy": {
+				Type:             schema.TypeString,
+				ValidateFunc:     validateIAMPolicyJson,
+				Required:         true,
+				DiffSuppressFunc: suppressEquivalentAwsPolicyDiffs,
+			},
+		},
+	}
+}
+
+func resourceAwsCloudSearchDomainCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).cloudsearchconn
+
+	input := cloudsearch.CreateDomainInput{
+		DomainName: aws.String(d.Get("domain_name").(string)),
+	}
+
+	output, err := conn.CreateDomain(&input)
+	if err != nil {
+		return fmt.Errorf("%s %q", err, d.Get("domain_name").(string))
+	}
+
+	d.SetId(*output.DomainStatus.ARN)
+	err = resourceAwsCloudSearchDomainUpdate(d, meta)
+
+	return err
+}
+
+func resourceAwsCloudSearchDomainRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).cloudsearchconn
+
+	input := cloudsearch.DescribeIndexFieldsInput{
+		DomainName: aws.String(d.Get("domain_name").(string)),
+	}
+
+	_, err := conn.DescribeIndexFields(&input)
+
+	return err
+}
+
+func resourceAwsCloudSearchDomainUpdate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).cloudsearchconn
+
+	err := updateScalingParameters(d, meta, conn)
+	if err != nil {
+		return err
+	}
+
+	updated, err := defineIndexFields(d, meta, conn)
+	if err != nil {
+		return err
+	}
+
+	err = updateAccessPolicy(d, meta, conn)
+	if err != nil {
+		return err
+	}
+
+	if updated {
+		_, err := conn.IndexDocuments(&cloudsearch.IndexDocumentsInput{
```

Comment:
Could you explain what this is used for please? not sure to understand. Thanks! 👍

Reply:
Hi,
as i understood if you change the index options you have to rebuild the search index.
http://docs.aws.amazon.com/cloudsearch/latest/developerguide/indexing.html